docs/csharp/language-reference/attributes/pseudo-attributes.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Attributes interpreted by the compiler: Pseudo-attributes"
+ms.date: 01/2/2025
+description: "Learn about attributes you can add to code that are written to IL as modifiers. These custom attributes aren't emitted as attributes in the compiled output."
+---
+# Custom attributes that generate flags or options in the Intermediate Language (IL) output
+
+You add these attributes to your code for the compiler to emit a specified Intermediate Language (IL) modifier. These attributes instruct the compiler to include the corresponding IL modifier in the output.
+
+| Attribute                                                                            | Modifier        | Comments  |
+|--------------------------------------------------------------------------------------|-----------------|-----------|
+| <xref:System.Runtime.InteropServices.ComImportAttribute?displayProperty=fullName>    | `import`        |           |
+| <xref:System.Runtime.InteropServices.DllImportAttribute?displayProperty=fullName>    | `pinvokeimpl`   | You can add options listed in the constructor. |
+| <xref:System.Runtime.InteropServices.FieldOffsetAttribute?displayProperty=fullName>  | `.field`        | This sets the field offset for memory layout.  |
+| <xref:System.Runtime.InteropServices.MarshalAsAttribute>                             | `marshal`       | You can set options listed in the constructor. |
+| <xref:System.Runtime.CompilerServices.MethodImplAttribute?displayProperty=fullName>  | `flag`          | Constructor arguments specify specific named flags such as `aggressiveinlining` or `forwardref`. These flags also specify the `native`, `managed`, or `optil` modifiers for the <xref:System.Runtime.CompilerServices.MethodCodeType?displayProperty=fullName> field. |
+| <xref:System.NonSerializedAttribute?displayProperty=fullName>                        | `notserialized` |           |
+| <xref:System.Runtime.InteropServices.OptionalAttribute?displayProperty=fullName>     | `[opt]`         |           |
+| <xref:System.Runtime.InteropServices.OutAttribute?displayProperty=fullName>          | `[out]`         |           |
```

Comment:
This one could have a note parallel to the note added for `in`, saying "Use the `out` modifier."